### PR TITLE
fix multithreading issue on macOS

### PR DESCRIPTION
**Pull request by scheffle**
fix multithreading issue by moving the clear color into the context state

### CODE REVIEW

File: src/nanovg_mtl.m [hunk 1]
```diff
@@ -502,11 +503,10 @@ static int mtlnvg__maxVertCount(const NVGpath* paths, int npaths,
 
   MTLRenderPassDescriptor *descriptor = \
       [MTLRenderPassDescriptor renderPassDescriptor];
-  descriptor.colorAttachments[0].clearColor = s_colorBufferClearColor;
-  descriptor.colorAttachments[0].loadAction = s_colorBufferLoadAction;
+  descriptor.colorAttachments[0].clearColor = mtl->clearColor;
+  descriptor.colorAttachments[0].loadAction = MTLLoadActionClear;
```
Comment (olliwang) [inline src/nanovg_mtl.m]: Always set to `MTLLoadActionClear` would reduce performance. Color shouldn't be cleared when it doesn't need to.

**Comment by scheffle** [comment]
Clear is now optional. mnvgClearWithColor needs to be called before nvgEndFrame to clear the color

File: src/nanovg_mtl.m [hunk 2]
```diff
@@ -1539,13 +1538,14 @@ void mnvgDeleteFramebuffer(MNVGframebuffer* framebuffer) {
   free(framebuffer);
 }
 
-void mnvgClearWithColor(NVGcolor color) {
+void mnvgClearWithColor (NVGcontext* ctx, NVGcolor color) {
```
Comment (olliwang) [inline src/nanovg_mtl.m]: The space behind the function name should be removed.

File: src/nanovg_mtl.m [hunk 3]
```diff
@@ -1082,6 +1079,8 @@ static int mtlnvg__renderCreate(void* uptr) {
   [backFaceStencilDescriptor release];
   [stencilDescriptor release];
 
+  mtl->clearBufferOnFlush = NO;
```
Comment (olliwang) [inline src/nanovg_mtl.m:1082]: The initial buffer should be cleared with `mtl->clearColor = MTLClearColorMake(0, 0, 0, 0)`.

Comment (scheffle) [reply src/nanovg_mtl.m:1082]: you mean:
```c
mtl->clearBufferOnFlush = YES;
mtl->clearColor = MTLClearColorMake(0, 0, 0, 0);
```
?

Comment (olliwang) [reply src/nanovg_mtl.m:1082]: Never mind. After thinking about it again. The current implementation is fine. 👍

**Comment by olliwang** [comment]
Thanks for the improvement. :)

**Pull request merged**